src/client/components/DashboardCalendar.tsx

```diff
@@ -1,20 +1,122 @@
-import React from "react";
-import { Card, TextField, Button, CardContent } from "@material-ui/core";
+import React, { useState } from "react";
+import moment from "moment"
 
-interface DashboardCalendarProps{
+import { Card, CardContent } from "@material-ui/core";
+import { Calendar, momentLocalizer, View, DateLocalizer } from "react-big-calendar"
 
+import { EventsModal } from "./Modals"
+
+import * as dates from "../../../node_modules/react-big-calendar/lib/utils/dates"
+
+import "!style-loader!css-loader!react-big-calendar/lib/css/react-big-calendar.css"
+
+{/* 
+====================================
+            READ ME
+Modules required for implementation:
+====================================            
+
+react-big-calendar
+moment
+
+*/}
+
+// Setup the localizer by providing the moment (or globalize) Object
+// to the correct localizer.
+let localizer = momentLocalizer(moment) // or globalizeLocalizer moment
+
+// instantiating the default view options to select
+const allViews: View[] = ['agenda', 'day', 'week', 'month'];
+
+// Relative date of the state properites
+interface Props {
+    localizer: DateLocalizer;
+}
+
+// Typescript implemenmtation of Calendar Events
+class CalendarEvent {
+    title: string;
+    allDay: boolean;
+    start: Date;
+    end: Date;
+    desc: string;
+    resourceId?: string;
+    tooltip?: string;
+
+    constructor(_title: string, _start: Date, _endDate: Date, _allDay?: boolean, _desc?: string, _resourceId?: string) {
+        this.title = _title;
+        this.allDay = _allDay || false;
+        this.start = _start;
+        this.end = _endDate;
+        this.desc = _desc || '';
+        this.resourceId = _resourceId;
+    }
 }
 
+
+const SelectableCalendar: React.FC<Props> = ( {localizer} ) => {
+
+    const [events, setEvents] = useState([{allDay:true, title: "Today"}] as CalendarEvent[]);
+    const [modalOpen, setModalOpen] = useState(false);
+    const [modalInputs, setModalInputs] = useState([]);
```

Comment:
modalInputs should really have a type instead of being an array of size 4, bad design. Look at using generics and interfaces with useState here: https://www.carlrippon.com/typed-usestate-with-typescript/

You can do the same above for the events useState